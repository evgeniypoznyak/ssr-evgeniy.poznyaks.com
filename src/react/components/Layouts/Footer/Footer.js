import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import BottomNavigation from '@material-ui/core/BottomNavigation';
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction';
import {Link} from 'react-router-dom';
import Avatar from '@material-ui/core/Avatar';
import {Paper} from '@material-ui/core';
import {Email, Phone} from '@material-ui/icons';

const useStyles = makeStyles({
    root: {
        marginTop: 20,
        // '@media (max-width: 960px)': {
        //     display: 'none',
        // },
    },
    menuWrapper: {
        display: 'flex',
        flexWrap: 'wrap',
        justifyContent: 'space-evenly',
    },
    avatar: {
        height: 25,
        width: 25,
    },
});

export default function Footer() {
    const AdapterLink = React.forwardRef((props, ref) => <Link innerRef={ref} {...props} />);
    const scrollToTop = () => {
        window.scrollTo(0, 0);
    };
    const classes = useStyles();
    const [value, setValue] = React.useState(null);

    return (
        <Paper className={classes.root}>
            <BottomNavigation
                value={value}
                onChange={(event, newValue) => {
                    setValue(newValue);
                }}
                showLabels
                className={classes.menuWrapper}
            >
                <BottomNavigationAction
                    onClick={scrollToTop}
                    label="Email Me"
                    component={AdapterLink}
                    to={'/contact-us/'}
                    icon={<Email/>}
                />
                <BottomNavigationAction
                    label="Call Me"
                    href="tel:518-772-8217"
                    icon={<Phone/>}
                />
                <BottomNavigationAction
                    color="inherit"
                    target="_blank"
                    label="My GuiHub"
                    href={'https://github.com/evgeniypoznyak/'}
                    icon={<Avatar className={classes.avatar} alt={'My GitHub'} src={'/assets/github-grey.jpg'}/>}
                />
                <BottomNavigationAction
                    color="inherit"
                    target="_blank"
                    label="My LinkedIn"
                    href={'https://www.linkedin.com/in/evgeniypoznyak/'}
                    icon={<Avatar alt={'My LinkedIn'} className={classes.avatar} src={'/assets/linkedin-grey.jpg'}/>}
                />
            </BottomNavigation>
        </Paper>

    );
}
