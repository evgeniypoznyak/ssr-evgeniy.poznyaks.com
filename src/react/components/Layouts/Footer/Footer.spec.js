import React from 'react';
import {mount, shallow} from 'enzyme';
import Footer from './Footer';

let wrapper;
let shallowWrapper;

beforeAll(() => {
    wrapper = mount(<Footer/>);
    shallowWrapper = shallow(<Footer/>);
});

describe('Footer', () => {
    it('should render correctly with enzyme', () => {
        expect(shallowWrapper.children.length).toBe(1);
        expect(shallowWrapper.getElement()).toMatchSnapshot();
    });
    it('should be text on the page', () => {
        expect(wrapper.text()).toMatch('Item');
    });
});
