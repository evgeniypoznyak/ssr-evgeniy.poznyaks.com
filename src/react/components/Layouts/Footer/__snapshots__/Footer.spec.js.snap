// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly with enzyme 1`] = `
<div
  className="makeStyles-root-1"
>
  <ForwardRef(WithStyles)
    color="inherit"
    position="static"
  >
    <ForwardRef(WithStyles)
      indicatorColor="primary"
      onChange={[Function]}
      textColor="primary"
      value={0}
      variant="fullWidth"
    >
      <ForwardRef(WithStyles)
        label="Item One"
      />
      <ForwardRef(WithStyles)
        label="Item Two"
      />
      <ForwardRef(WithStyles)
        label="Item Three"
      />
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
</div>
`;
